### PR TITLE
Improve GitHub PR block

### PR DESCRIPTION
**Pull request by teenoh**
## 🌟 What is the purpose of this PR?

This adds improvements to the Github PR Overview block and also updates it to use Þ v0.2 methods.

## 🔗 Related links

- [Implement block polish: GitHub PR viewer](https://app.asana.com/0/1202482456346912/1202208104881132) _(internal)_
- [Update GitHub PR block to 0.2](https://app.asana.com/0/1202482456346912/1202418951550826) _(internal)_

## 🔍 What does this change?

- Updates the block to match the new design ([internal link](https://www.figma.com/file/T10d2P2iNb1LdRC5a9J7vA/BP-Block-Week-Designs?node-id=83%3A2213))
- Updates the block to work with Þ v0.2 methods
- Updates entities/entityTypes in `example-graph.json` to match the new structure in Þ v0.2
- Updates how global state is managed to use useReducer.
- Simplifies how entities/entityTypes are fetched. We previously had to make multiple calls to fetch entities but can now do it in one call
- Adds UI for loading/error state

## ❓ How to test this?

1.  Checkout the branch
1.  Test the block as normal

## 📹 Demo

https://user-images.githubusercontent.com/16928688/176672479-2af5b500-32f3-4491-9277-aa75f8920fa2.mp4

### CODE REVIEW

File: packages/blocks/github-pr-overview/src/app.tsx
Comment (teenoh) [inline packages/blocks/github-pr-overview/src/app.tsx:30]: This shows how we can customise the theme exported from our design system package

File: packages/blocks/github-pr-overview/src/entity-aggregations.ts
Comment (Alfred-Mountfield) [reply packages/blocks/github-pr-overview/src/entity-aggregations.ts:73]: It was honestly just something I did during development to experiment with the interface. I am not sure if we want to possibly get ALL things in one call because that seems like it might be slower as opposed to doing multiple promises 'in parallel', and also I'm a little worried about the implication of requesting a tonne of data in one request if backends end up limiting the size of a request due to their memory constraints or something.

Comment (teenoh) [reply packages/blocks/github-pr-overview/src/entity-aggregations.ts:73]: Sorry I didn't make this clearer.  I went with the approach of fetching all things because the existing setup did that. I figure we'd want to make this optimal later on by initially fetching a small amount of data and getting more if the user wants that (could be via a load more button or on scroll). Decided to leave that out since it's out of scope for now. 

I didn't consider the fact that multiple api calls in 'parallel' will be faster than one call or backends could limit the sizing of requests. I do feel we'd eventually rewrite this since the current behaviour of fetching "ALL PRs" before a user can use the block isn't a great experience.

Happy to revert though if you want me to

File: packages/blocks/github-pr-overview/src/app.tsx
Comment (Alfred-Mountfield) [inline packages/blocks/github-pr-overview/src/app.tsx:14]: Capitalization seems inconsistent, do we have a standard in HASH for this yet? I really am not a fan of capitalized acronyms in PascalCase, especially when you do things like try to convert it to another case (which we do with some data in this block..)

[Rust for example](https://rust-lang.github.io/api-guidelines/naming.html) says 
> In UpperCamelCase, acronyms and contractions of compound words count as one word: use Uuid rather than UUID, Usize rather than USize or Stdin rather than StdIn. In snake_case, acronyms and contractions are lower-cased: is_xid_start.

File: packages/blocks/github-pr-overview/src/app.tsx
Comment (Alfred-Mountfield) [inline packages/blocks/github-pr-overview/src/app.tsx:140]: Why is this necessary? Did 0.2 get implemented with the Ids being objects?

File: packages/blocks/github-pr-overview/src/app.tsx
Comment (Alfred-Mountfield) [inline packages/blocks/github-pr-overview/src/app.tsx:231]: Is this todo still necessary? It seems weirdly placed

File: packages/blocks/github-pr-overview/src/entity-aggregations.ts
Comment (Alfred-Mountfield) [inline packages/blocks/github-pr-overview/src/entity-aggregations.ts:17]: It seems weird to me to keep the `numPages` argument if we're just using it as a multiplier for some const value. I think we should either swap to having a `num_items` parameter, or to making `itemsPerPage` a top-level const on its own.

File: packages/blocks/github-pr-overview/src/pull-request-selector/index.tsx
Comment (Alfred-Mountfield) [inline packages/blocks/github-pr-overview/src/pull-request-selector/index.tsx:83]: This seems a little inconsistent with the naming of the block. I don't really mind which way we go but I think we should either say something like 
> Select a GitHub pull request to show an Overview of 

(perhaps better phrased)

Or we should rename the block back to `github-pr-timeline` or something. I think it's more than just a timeline personally, and I think we might expand it with more things in the future so I'm inclined towards overview.

Comment (Alfred-Mountfield) [inline packages/blocks/github-pr-overview/src/pull-request-selector/index.tsx:83]:
```suggestion
          Select a GitHub pull request to create a timeline
```

File: packages/blocks/github-pr-overview/package.json
Comment (teenoh) [reply packages/blocks/github-pr-overview/package.json:13]: It will get changed prior to merging. Added it there so it's easy to test the block on BP Hub => https://github.com/blockprotocol/blockprotocol/pull/407

File: packages/blocks/github-pr-overview/src/app.tsx
Comment (teenoh) [reply packages/blocks/github-pr-overview/src/app.tsx:140]: This is an object. I renamed `selectedPullRequest` (which is an object) to `selectedPullRequestId` because I felt the name conflicts with the `pullRequest` property we have in the local state. 

https://github.com/hashintel/hash/blob/90859ac19b414dda9db476e6b5e2857e25bbeb6c/packages/blocks/github-pr-overview/src/app.tsx#L84-L85

https://github.com/hashintel/hash/blob/90859ac19b414dda9db476e6b5e2857e25bbeb6c/packages/blocks/github-pr-overview/src/app.tsx#L87-L100

Thinking about it, I guess `selectedPullRequestId` can be assumed to be a string. Happy to revert back to `selectedPullRequest` if you prefer that

File: packages/blocks/github-pr-overview/src/entity-aggregations.ts
Comment (teenoh) [reply packages/blocks/github-pr-overview/src/entity-aggregations.ts:17]: I agree. Will update

File: packages/blocks/github-pr-overview/src/app.tsx
Comment (teenoh) [reply packages/blocks/github-pr-overview/src/app.tsx:231]: It no longer is. Thanks for catching

File: packages/blocks/github-pr-overview/package.json
Comment (Alfred-Mountfield) [reply packages/blocks/github-pr-overview/package.json:13]: Makes sense, leaving this open so we can resolve just before merging

File: packages/blocks/github-pr-overview/src/app.tsx
Comment (Alfred-Mountfield) [reply packages/blocks/github-pr-overview/src/app.tsx:140]: Gotcha, I was looking at this through GitHub so I wasn't benefiting from intellisense so I didn't easily see the type. I understand the rename, I think the issue I see is that we have actual PullRequest entities, and this is just an identifier for one. So perhaps we can call it `selectedPullRequestIdentifier` or something? Maybe it's just that `id` tends to have connotations of Strings.

I'm not too bothered, though, also more than happy to leave as is because when using an IDE this would be a lot clearer.

File: packages/blocks/github-pr-overview/src/app.tsx
Comment (teenoh) [reply packages/blocks/github-pr-overview/src/app.tsx]: updated the casing here => https://github.com/hashintel/hash/pull/705/commits/3d98977e4140b53471f95443e4edf4f8e6172868

File: packages/blocks/github-pr-overview/src/app.tsx
Comment (teenoh) [reply packages/blocks/github-pr-overview/src/app.tsx]: Updated here => https://github.com/hashintel/hash/pull/705/commits/ac1ebefdd4cd146a393b31476354710c2a52c72e#diff-823b1aa85546193960a17d9c44f706004c4594ceb1214dfd64b48b4b3d22ddf7

**Review by Alfred-Mountfield — COMMENTED** [review]
Happy to approve when we update the URL in the package.json

**Review by CiaranMn — APPROVED** [review]

**Pull request merged**
